Sources/BadondeKit/Output.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+import CryptoSwift
+import Git
+import GitHub
+import Sugar
+
+public struct Output: Codable {
+	public struct PullRequest: Codable {
+		public var title: String
+		public var headBranch: String
+		public var baseBranch: String
+		public var body: String?
+		public var assignees: [String]
+		public var labels: [Label]
+		public var milestone: Milestone?
+		public var isDraft: Bool
+	}
+
+	public struct AnalyticsData: Codable {
+		public var info: [String: AnyCodable]
+	}
+
+	public var pullRequest: PullRequest
+	public var analyticsData: AnalyticsData
+}
+
+extension Output: CustomStringConvertible {
+	public var description: String {
+		let indentation: (Int) -> String = { String(repeating: " ", count: $0) }
+
+		let assignees = !pullRequest.assignees.isEmpty ? "\n" + pullRequest.assignees.map { indentation(6) + $0 }.joined(separator: "\n") : "<none>"
+		let labels = !pullRequest.labels.isEmpty ? "\n" + pullRequest.labels.map { indentation(6) + $0.name }.joined(separator: "\n") : "<none>"
+
+		let pullRequestDescription = """
+		Pull request:
+		   title: \(pullRequest.title)
+		   headBranch: \(pullRequest.headBranch)
+		   baseBranch: \(pullRequest.baseBranch)
+		   body: \(pullRequest.body ?? "<none>")
+		   assignees: \(assignees)
+		   labels: \(labels)
+		   milestone: \(pullRequest.milestone?.title ?? "<none>")
+		   isDraft: \(pullRequest.isDraft)
+		"""
+
+		guard !analyticsData.info.isEmpty else {
+			return pullRequestDescription
+		}
+
+		let analyticsDataInfo = analyticsData.info
+			.map { indentation(3) + "\($0): \($1)" }
+			.joined(separator: "\n")
+
+		return pullRequestDescription + "\n\n" + """
+		Analytics data:
+		\(analyticsDataInfo)
+		"""
+	}
+}
+
+extension Output {
+	public static func path(for repository: Repository) -> URL {
+		let repositoryPathSha1 = repository.topLevelPath.path.sha1()
+		return FileManager.default.temporaryDirectory.appendingPathComponent("badonde-output-\(repositoryPathSha1).json")
+	}
+}
+
+var output: Output!
```

Comment:
<!--
  1 failure:  Implicitly unwrap...
  0 warning: 
  
  
  DangerID: danger-id-default;
  File: Sources/BadondeKit/Output.swift;
  Line: 68;
-->
- :no_entry_sign: Implicitly unwrapped optionals should be avoided when possible. (`implicitly_unwrapped_optional`)